File: Excel/500-599/575/575 Challenge.py
import pandas as pd

df = pd.read_excel("575 List Above and Below Average Salary.xlsx", usecols="A:B", skiprows=1, nrows=8)
test = pd.read_excel("575 List Above and Below Average Salary.xlsx", usecols="D:E", skiprows=1, nrows=12)

df = df.assign(Names=df['Names'].str.split(', ')).explode('Names')
df[['Name', 'Salary']] = df['Names'].str.split('-', expand=True)
df['Salary'] = pd.to_numeric(df['Salary'], errors='coerce')
avg_salary = df['Salary'].mean()

df['AboveAvg'] = df['Salary'].apply(lambda x: '>=Average' if x >= avg_salary else '< Average')
df['nr'] = df.groupby('AboveAvg').cumcount() + 1
df['Names'] = df['Dept'] + '-' + df['Name']

result = df.pivot(index='nr', columns='AboveAvg', values='Names').reset_index(drop=True)[['< Average', '>=Average']]
result.columns.name = None

print(result.equals(test)) # True

File: Excel/500-599/575/575 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/575 List Above and Below Average Salary.xlsx"
input = read_excel(path, range = "A2:B9")
test  = read_excel(path, range = "D2:E13")

result = input %>%
  separate_rows(Names, sep = ", ") %>%
  separate(Names, into = c("Name", "Salary"), sep = "-") %>%
  mutate(AvgSalary = mean(as.numeric(Salary), na.rm = T),
         AboveAvg = ifelse(Salary >= AvgSalary, ">= Average", "< Average")) %>%
  mutate(nr = row_number(), .by = AboveAvg,
         Names  = paste0(Dept,"-",Name)) %>%
  select(Names, AboveAvg, nr) %>%
  pivot_wider(names_from = AboveAvg, values_from = Names) %>%
  select(`< Average`, `>= Average`) 

all.equal(result, test, check.attributes = F)
#> [1] TRUE